keyboards/keychron/q1/rev_0107/rules.mk

```diff
@@ -0,0 +1,31 @@
+# MCU name
+MCU = STM32L433
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes        # Enable Encoder
+DIP_SWITCH_ENABLE = yes
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = CKLED2001
+EEPROM_DRIVER = transient
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+
+# custom matrix setup
+CUSTOM_MATRIX = lite
+
+SRC += matrix.c
```

Comment:
```suggestion
```

Reply:
Sorry, not sure how I lost the [matrix.c](https://github.com/Keychron/qmk_firmware/blob/master/keyboards/keychron/q1/matrix.c)(put in q1 home directly) and didn't upload it. To be honest, we used an I/O extension IC 74HC595 for some types of Q1, so please review it again. Thanks a lot.